clang/lib/CIR/CodeGen/CIRGenCXX.cpp

```diff
@@ -20,6 +20,32 @@
 using namespace clang;
 using namespace cir;
 
+static void buildDeclInit(CIRGenFunction &CGF, const VarDecl *D,
+                          Address DeclPtr) {
+  assert((D->hasGlobalStorage() ||
+          (D->hasLocalStorage() &&
+           CGF.getContext().getLangOpts().OpenCLCPlusPlus)) &&
+         "VarDecl must have global or local (in the case of OpenCL) storage!");
+  assert(!D->getType()->isReferenceType() &&
+         "Should not call buildDeclInit on a reference!");
+
+  QualType type = D->getType();
+  LValue lv = CGF.makeAddrLValue(DeclPtr, type);
+
+  const Expr *Init = D->getInit();
+  switch (CIRGenFunction::getEvaluationKind(type)) {
+  case TEK_Aggregate:
+    CGF.buildAggExpr(
+        Init, AggValueSlot::forLValue(lv, AggValueSlot::IsDestructed,
+                                      AggValueSlot::DoesNotNeedGCBarriers,
+                                      AggValueSlot::IsNotAliased,
+                                      AggValueSlot::DoesNotOverlap));
+    return;
```

Comment:
Please also add cases+asserts for `TEK_Scalar and TEK_Complex` or make the default assert more meanigful.